core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java

```diff
@@ -2086,6 +2087,49 @@ public int getMaxSIRepairLimit(String dbName, String tableName) {
     return Math.abs(Integer.parseInt(thresholdValue));
   }
 
+  /**
+   * The below method returns the time(in milliseconds) for which timestamp folder retention in
+   * trash folder will take place.
+   */
+  public long getTrashFolderRetentionTime() {
+    long milliSecondsInADay = TimeUnit.DAYS.toMillis(1);
+    return (long)validateTrashFolderRetentionTime() * milliSecondsInADay;
+  }
+
+  /**
+   * The below method returns the time(in days) for which timestamp folder retention in trash
+   * folder will take place
+   */
+  private int validateTrashFolderRetentionTime() {
```

Comment:
invoke this method in validateAndLoadDefaultProperties

Reply:
done